modules/azure-flexible-server-postgresql/data.tf

```
@@ -0,0 +1,43 @@
+# Data section
+#https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_group
+data "azurerm_resource_group" "resource_group" {
+  name = var.resource_group
+}
+
+#https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/subnet
+data "azurerm_subnet" "subnet" {
+  name                 = var.subnet.name
+  virtual_network_name = local.virtual_network_name
+  resource_group_name  = local.resource_group_name
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/private_dns_zone
+data "azurerm_private_dns_zone" "dns_private_zone" {
+  name                = var.dns_private_zone.name
+  resource_group_name = local.resource_group_name
+}
+
+data "azurerm_resources" "subnet" {
```

```suggestion
data "azurerm_resources" "subnet" {
```
Esto non está buscando subnet, está buscando a vnet